tests/pages/remoteInstancesPage.js

```diff
@@ -163,18 +171,39 @@ module.exports = {
     return this;
   },
 
+  async fillTLS(file, field) {
+    const certificateData = await I.verifyCommand(`cat /tmp/ssl/pmm-ui-tests/testdata/mysql/ssl-cert-scripts/certs/${file}`);
+
+    I.waitForVisible(field, 30);
+    I.fillField(field, certificateData);
+  },
+
+  fillEnvironmentAndCluster(serviceName) {
```

Comment:
is there a need to have this logic just to have environment and cluster fields filled?

Reply:
Yes, I want to use all the steps to fill fields for MySQL but the only difference is the cluster and environment. And having a Switch statement inside another switch statement looks weird. :)

Reply:
As I recall those fields are not required. Do you use those fields to filter or what? If not, then I don't really see a reason to add more complexity here @puneet0191

Reply:
I don't use them now. I will explain problem which can happen: The MySQL with TLS will be added, and MySQL without TLS not. Then we have a test to check QAN page for MySQL without TLS and we are checking that by cluster or environment. Then the test for check QAN will be false positive.

Reply:
Ok, but can you filter QAN by service name?

Reply:
I can :) It refactoring of older tests.

Reply:
env and cluster fields only needed to check if they are rightly available in QAN filter section, but if you want to check QAN queries you can use the service name also